src/Subscriptions/LegacyListeners/DispatchGiveSubscriptionPreCreate.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Give\Subscriptions\LegacyListeners;
+
+use Give\Helpers\Hooks;
+use Give\Subscriptions\Models\Subscription;
+
+class DispatchGiveSubscriptionPreCreate
+{
+    /**
+     * @unreleased
+     *
+     * @param  Subscription  $subscription
+     * @return void
+     */
+    public function __invoke(Subscription $subscription)
+    {
+		$args = [
+			'customer_id'          => $subscription->donorId,
+			'period'               => $subscription->period->getValue(),
+			'frequency'            => $subscription->frequency,
+			'initial_amount'       => $subscription->amount,
+			'recurring_amount'     => $subscription->amount,
+			'recurring_fee_amount' => $subscription->feeAmount,
+			'bill_times'           => $subscription->installments,
+			'parent_payment_id'    => 0,
+			'form_id'              => $subscription->donationFormId,
+			'created'              => null,
+			'expiration'           => $subscription->expiration(),
+			'status'               => $subscription->status->getValue(),
+			'profile_id'           => $subscription->gatewaySubscriptionId,
+		];
+
```

Comment:
Tabs?

Reply:
reformatted file, not sure what happened there e9d2a7aa